src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsNavigationSource.cs

```diff
@@ -130,11 +130,20 @@ public IEnumerable<IEdmNavigationPropertyBinding> FindNavigationPropertyBindings
         {
             if (!navigationProperty.ContainsTarget)
             {
-                IEnumerable<IEdmNavigationPropertyBinding> navigationBindingList = null;
+                IList<IEdmNavigationPropertyBinding> navigationBindingList = null;
 
                 if (!navigationPropertyBindingCache.TryGetValue(navigationProperty, out navigationBindingList))
                 {
-                    navigationBindingList = this.NavigationPropertyBindings.Where(targetMapping => targetMapping.NavigationProperty == navigationProperty);
+                    navigationBindingList = new List<IEdmNavigationPropertyBinding>();
+
+                    foreach (IEdmNavigationPropertyBinding targetMapping in this.NavigationPropertyBindings)
+                    {
+                        if (targetMapping.NavigationProperty == navigationProperty)
+                        {
+                            navigationBindingList.Add(targetMapping);

```

Comment:
Add range should be faster in most cases as it uses a faster array copy method for the items.

Reply:
You mean add range with a where to check for navigationproperty? that would be slower right?